src/common/arrstr.cpp

```diff
@@ -759,47 +756,48 @@ struct wxStringFragment
 
 wxStringFragment GetFragment(wxString& text)
 {
-    static const wxRegEx reSpaceOrPunct(wxS("^([[:space:]]|[[:punct:]])+"));
-    // Limit the length to make sure the value will fit into a wxUint64
-    static const wxRegEx reDigit(wxS("^[[:digit:]]{1,19}"));
-    static const wxRegEx reLetterOrSymbol("^[^[:space:]|[:punct:]|[:digit:]]+");
-
     if ( text.empty() )
         return wxStringFragment();
 
-    wxStringFragment fragment;
-    size_t           length = 0;
+    wxStringFragment   fragment;
+    wxString::iterator it;
 
-    // In attempt to minimize the number of wxRegEx.Matches() calls,
-    // try to do them from the most expected to the least expected
-    // string fragment type.
-    if ( reLetterOrSymbol.Matches(text) )
-    {
-        if ( reLetterOrSymbol.GetMatch(NULL, &length) )
-        {
-            fragment.type = wxStringFragment::LetterOrSymbol;
-            fragment.text = text.Left(length);
-        }
-    }
-    else if ( reDigit.Matches(text) )
+    for ( it = text.begin(); it != text.end(); ++it )
     {
-        if ( reDigit.GetMatch(NULL, &length) )
+        const wxChar           ch = *it;
```

Comment:
`ch` should probably be `wxUniChar` instead of `wxChar`, to make this actually work in UTF-8 builds....